src/librustc_privacy/lib.rs

```diff
@@ -92,7 +93,13 @@ impl<'a, 'tcx> EmbargoVisitor<'a, 'tcx> {
         let ty_def_id = match self.tcx.type_of(item_def_id).sty {
             ty::Adt(adt, _) => adt.did,
             ty::Foreign(did) => did,
-            ty::Dynamic(ref obj, ..) => obj.principal().def_id(),
+            ty::Dynamic(ref obj, ..) => {
+                if let Some(principal_def_id) = obj.principal_def_id() {
+                    principal_def_id
+                } else {
+                    return Some(AccessLevel::Public)
```

Comment:
What this code (and other uses of `ty::Dynamic` in rustc_privacy) really wants is to treat all the bounds (principal and not principal) uniformly and find minimum or maximum on them.
It's using principal because when it was written only `Send` and `Sync` could be non-principal components, and they could always be considered public.
Now things changed, non-principal components can be private auto traits, so this behavior is actually a bug.

Reply:
Which of them is it - minimum or maximum?

Reply:
Yes, this is a pre-existing issue.
I'll try to come up with some tests.

Reply:
Fixed in https://github.com/rust-lang/rust/pull/56878